filters/ratelimit/ratelimit.go

```diff
@@ -379,6 +392,20 @@ func getDurationArg(a interface{}) (time.Duration, error) {
 	return time.Duration(i) * time.Second, err
 }
 
+func getStatusCodeArg(args []interface{}, index int) (int, error) {
+	// status code arg is optional so we return 0 but no error
```

Comment:
Simply return `http.StatusTooManyRequests` and don't bother checking zero in `Request`

Reply:
@AlexanderYastrebov I can't since the argument is used just when creating `ratelimit` and `clusterRateLimit` filters, for the other filters the `statusCode` filter property will be equal to 0.

Reply:
to me it still seems possible that what @AlexanderYastrebov suggested could work. In general, how it should work:

- accept the optional parameter only where we use it
- don't accept more parameters

This should be possible, because the spec type has a typ field marking which concrete filter we're using, and each filter has it's own init function. Do I miss something?

Reply:
> accept the optional parameter only where we use it 
The criteria is met with the current implementation. The optional param can be used just with "non-client rate limit" filters.
> don't accept more parameters 
I will update the code to have a more strict check as you suggested.

About
> Simply return `http.StatusTooManyRequests` and don't bother checking zero in `Request`.

I am using the filter struct to hold the status code which will be used for a limited response. Since we accept the optional parameter just on "non-client rate limit" filters for client-based rate limit filters the status code will be equal to 0. If you prefer, I can instead of checking zero in the `Request` method initialize the statusCode for each filter to the default (`http.StatusTooManyRequests`).

Reply:
> If you prefer, I can instead of checking zero in the Request method initialize the statusCode for each filter to the default (http.StatusTooManyRequests).

Yes, that's kind of also how I imagined it. Thx!

Reply:
@rbarilani Sorry, I've jumped in with a comment but then realized it is a draft so I've decided to look later.
What I've meant indeed is to have a status field initialized with default backward-compatible 429 and override it with optinal filter argument. Since status code does not change through the filter lifetime it is not necessary to do any runtime check in the `Request()` hot path.